### DISCUSSION TITLE
Vercel Edge, waitUntil

### DISCUSSION THREAD

**Discussion by enesakar** [vercel/community#905] [Help] [answered]
Hello

I am trying to use Edge middleware. I do not want to use `await` inside the middleware but rather the runtime to wait until all promises to be completed. There is `waitUntil` in cloudflare workers for that. How can I do this in Vercel Edge middleware?

https://developers.cloudflare.com/workers/runtime-apis/fetch-event/#waituntil

**Comment by amyegan** [comment, marked as answer]
Hi @enesakar! The `waitUntil` method is actually part of the Service Worker API, so you are not tied to Cloudflare to use it. You can read more about it here: https://developer.mozilla.org/en-US/docs/Web/API/ExtendableEvent/waitUntil

That said, I'm not sure whether it will work with Edge Middleware. I have not tried it and do not have a service workers example app set up to test it. It may be more appropriate to use [Edge Functions](https://vercel.com/docs/concepts/functions/edge-functions) or [Serverless Functions](https://vercel.com/docs/concepts/functions/serverless-functions), depending on what you are building.
